prinz/src/main/scala/pl/touk/nussknacker/prinz/model/Model.scala

```diff
@@ -1,14 +1,29 @@
 package pl.touk.nussknacker.prinz.model
 
-trait Model {
+import pl.touk.nussknacker.prinz.model.SignatureProvider.ProvideSignatureResult
 
-  def getName: ModelName
+trait Model {
 
-  def getVersion: ModelVersion
+  def getMetadata: ModelMetadata = ModelMetadata(getName, getVersion, getSignature, getParameterDefinition)
 
   def toModelInstance: ModelInstance
 
+  protected val signatureOption: ProvideSignatureResult
+
   override def toString: String = s"Model $getName version: $getVersion"
+
+  // defines input of scored external model
+  protected final def getSignature: ModelSignature = signatureOption match {
+    case Right(value) => value
+    case Left(exception) => throw SignatureNotFoundException(exception)
+  }
+
+  // defines input of enricher model
+  protected def getParameterDefinition: ModelSignature = getSignature
```

Comment:
We could rethink proxying in the future. It's an optional layer, and I'm not very happy with doubling `getSignature`. Maybe some decorator around `Model`/`SignatureProvider`?

Reply:
Later I noticed you actually create a child class for proxied models, it could just return a different signature in `getMetadata`.

Reply:
it returns actually different  signature in `getMetadata` but as a field of `parameterDefinition`
These get* methods are to be overriden and then model automatically wraps these metadata to Metadata object which is visible in public API

Reply:
I meant giving a different signature in `ModelMetadata.signature`, exactly the thing that you described in #177.